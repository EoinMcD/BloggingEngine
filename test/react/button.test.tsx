import * as React from "react";
import MyButton from "../../app/javascript/components/button";
import { expect, test } from "@jest/globals";
import { render, RenderResult, getByRole } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/jest-dom";

const renderButton = (path = "/signup", name = "Register"): RenderResult => {
  return render(<MyButton path={path} name={name} />);
};

test("Component renders as expected", () => {
  const { container } = renderButton();
  expect(container).toMatchSnapshot();
});

test("Register button has the right link", () => {
  const { container } = renderButton();
  expect(getByRole(container, "link")).toHaveProperty(
    "href",
    window.location.href + "signup"
  );
});

test("Login button has the right link", () => {
  const { container } = renderButton("/login/index");
  expect(getByRole(container, "link")).toHaveProperty(
    "href",
    window.location.href + "login/index"
  );
});
