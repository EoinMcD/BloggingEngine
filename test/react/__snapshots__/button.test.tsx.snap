// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    href="/signup"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      Register
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</div>
`;
